pytest_catchlog.py

```diff
@@ -162,59 +190,96 @@ def record_tuples(self):
 
             (logger_name, log_level, message)
         """
-        return [(r.name, r.levelno, r.getMessage()) for r in self.records()]
-
-    def set_level(self, level, logger=None):
-        """Sets the level for capturing of logs.
-
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
-        """
+        return [(r.name, r.levelno, r.getMessage()) for r in self.buffer]
 
-        obj = logger and logging.getLogger(logger) or self.handler
-        obj.setLevel(level)
+    @property
+    def buffer(self):
+        return self._handler.buffer
 
-    def at_level(self, level, logger=None):
-        """Context manager that sets the level for capturing of logs.
+    @property
+    def lines(self):
+        return [r.getMessage() for r in self.buffer]
 
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
+    def __init__(self, handler):
+        # Do not call LineMatcher.__init__(),
+        # it attempts to assign to self.lines, which is a property.
```

Comment:
I think it depends on whenever `LineMatcher` has been declared as public API whenever it should be reused or not. Duplicating code is something I'm not very fond of. And since it actually has been [documented](http://pytest.org/latest/writing_plugins.html?highlight=linematcher#_pytest.pytester.LineMatcher) I would say it's okaish unless you still would say it's still a big no. Your words have more weight compared to mine in this field since you're the one with more experience in py.test development.